pkg/descheduler/strategies/pod_lifetime.go

```diff
@@ -78,15 +88,24 @@ func PodLifeTime(ctx context.Context, client clientset.Interface, strategy api.D
 	evictable := podEvictor.Evictable(evictions.WithPriorityThreshold(thresholdPriority))
 
 	filter := evictable.IsEvictable
-	if strategy.Params.PodLifeTime.PodStatusPhases != nil {
-		filter = func(pod *v1.Pod) bool {
-			for _, phase := range strategy.Params.PodLifeTime.PodStatusPhases {
-				if string(pod.Status.Phase) == phase {
-					return evictable.IsEvictable(pod)
+
+	states := getPodLifeTimeStates(strategy.Params.PodLifeTime)
```

Comment:
`validatePodLifeTimeParams` could return the states so that this doesn't need to be its own function called twice

Reply:
True. It would be somewhat odd for the validate params to return only a subset of the params.

Reply:
Applied